tool/tsh/tsh.go

```diff
@@ -3132,9 +3136,29 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 	// addresses from the proxy (this is what Ping does).
 	if cf.IdentityFileIn != "" {
 		log.Debug("Pinging the proxy to fetch listening addresses for non-web ports.")
-		if _, err := tc.Ping(cf.Context); err != nil {
+		pr, err := tc.Ping(cf.Context)
+		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		tc.LoadAllCAs = pr.Auth.LoadAllHostCAs
+
+		if tc.LoadAllCAs {
```

Comment:
Nit: Add a comment here explaining that now that we've pinged the proxy and determined whether the client needs to load all CAs, we need to update host key callback and TLS config on the client from what we set up above.